proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+// of the Mintter Site to proxy the request to.
+//
+// TODO(burdiyan): Document name of the required metadata header.
+service WebSite {
+  // Creates a new invite token for registering a new member.
+  // Protected. Must require authentication.
+  rpc CreateInviteToken(CreateInviteTokenRequest) returns (InviteToken);
+
+  // Redeems a previously created invite token to register a new member.
+  rpc RedeemInviteToken(RedeemInviteTokenRequest) returns (RedeemInviteTokenResponse);
+
+  // Gets public-facing site information.
+  rpc GetSiteInfo(GetSiteInfoRequest) returns (SiteInfo);
+
+  // Updates public-facing site information. Doesn't support partial updates,
+  // hence all the fields must be provided.
+  // Protected. Must require authentication.
+  rpc UpdateSiteInfo(UpdateSiteInfoRequest) returns (SiteInfo);
+
+  // Lists registered members on the site.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Gets information about a specific member.
+  // May be protected or public depending on the privacy policies of the web site.
+  rpc GetMember(GetMemberRequest) returns (Member);
+
+  // Deletes an existing member.
+  // Protected. Must require authentication.
+  rpc DeleteMember(DeleteMemberRequest) returns (google.protobuf.Empty);
+
+  // Publishes a Mintter Document on the remote web server along with any supporting materials.
+  rpc PublishDocument(PublishDocumentRequest) returns (PublishDocumentResponse);
+
+  // Unpublishes a previously published Document.
+  rpc UnpublishDocument(UnpublishDocumentRequest) returns (UnpublishDocumentResponse);
+
+  // list all the published documents
+  rpc ListWebPublications(ListWebPublicationsRequest) returns (ListWebPublicationsResponse);
+}
+
+// Request to create a new invite token.
+message CreateInviteTokenRequest {
+  // Required. The role to be associated with the new member after invite is redeemed.
+  // The server might be configured to limit invites for certain roles, but it's not
+  // restricted in this API definition.
+  Member.Role role = 1;
+
+  // Optional. The timestamp after which the invite token will expire if not redeemed.
+  // If not provided, the server will decide the deadline based on the internal configuration.
+  google.protobuf.Timestamp expire_time = 2;
+}
+
+// Request to redeem an invite token. This is the most security-sensitive request,
+// because it allows to register new members on the site. The server must obtain
+// and verify the relation of the communicating peer with a corresponding Mintter Account.
+// After the token is redeemed, a new member on the site must be created, associating
+// the Mintter Account of the caller with the role invite token was created for.
+message RedeemInviteTokenRequest {
+  // Value of the invite token.
+  string token = 1;
+
+  // TODO: might not need these, because can be inferred from the communication channel.
```

Comment:
I would remove both `account_id` and `device_id`. No need to double check, And we did not even use `device_id`  in the first place since all the authentication depends on accountIDs rather than individual devices.

Reply:
Do it. I don’t mind. We can add later if we need anything.